sampleChangerApp/src/converter.cpp

```diff
@@ -231,52 +244,101 @@ std::string converter::get_available_in_slot(std::string slot)
+
+        slotData& slot = it->second;
+        std::vector<std::pair<std::string, std::vector<std::pair<std::string, samplePosn>>>>::iterator iter = std::find_if(v_racks.begin(), v_racks.end(),
+            [&](std::pair<std::string, std::vector<std::pair<std::string, samplePosn>>> pair) {return pair.first == slot.rackType; });
+        if (iter == v_racks.end()) {
             errlogPrintf("sampleChanger: Unknown rack type '%s' of slot %s\n", slot.rackType.c_str(), slot.name.c_str());
             return 1;
         }
         else {
-            for ( std::map<std::string, samplePosn>::iterator it2 = iter->second.begin() ; it2!=iter->second.end() ; it2++ ) {
+            for (std::vector<std::pair<std::string, samplePosn>>::iterator it2 = iter->second.begin(); it2 != iter->second.end(); it2++) {
                 std::string full_position_name = it2->second.name + slot.sampleSuffix;
-                m_positions_for_each_slot[ALL_POSITIONS_NAME].push_back(full_position_name);
-                m_positions_for_each_slot[slot.name].push_back(full_position_name);
-                m_slot_for_each_position[full_position_name] = slot.name;
 
-                if ( m_dims==1 ) {
-                    fprintf(fpOut, "%s %f\n", full_position_name.c_str(), it2->second.x+slot.x+slot.xoff);                    
+                std::vector<std::pair<std::string, std::list<std::string>>>::iterator pos_it =
+                    std::find_if(v_positions_for_each_slot.begin(), v_positions_for_each_slot.end(),
+                        [&](std::pair<std::string, std::list<std::string>> pair) {return pair.first == ALL_POSITIONS_NAME; });
+                if (pos_it == v_positions_for_each_slot.end())
+                {
+                    v_positions_for_each_slot.push_back(std::make_pair(ALL_POSITIONS_NAME, std::list<std::string>()));
+                }
+                (std::find_if(v_positions_for_each_slot.begin(), v_positions_for_each_slot.end(),
+                    [&](std::pair<std::string, std::list<std::string>> pair) {return pair.first == ALL_POSITIONS_NAME; }))
+                    ->second.push_back(full_position_name);
+
+                std::vector<std::pair<std::string, std::list<std::string>>>::iterator pos_it2 =
+                    std::find_if(v_positions_for_each_slot.begin(), v_positions_for_each_slot.end(),
+                        [&](std::pair<std::string, std::list<std::string>> pair) {return pair.first == slot.name; });
+                if (pos_it2 == v_positions_for_each_slot.end())
+                {
+                    v_positions_for_each_slot.push_back(std::make_pair(slot.name, std::list<std::string>()));
+                }
+                (std::find_if(v_positions_for_each_slot.begin(), v_positions_for_each_slot.end(),
+                    [&](std::pair<std::string, std::list<std::string>> pair) {return pair.first == slot.name; }))
+                    ->second.push_back(full_position_name);
+
+                try
```

Comment:
Why are we only catching errors for the final insert and not the others? What sort of errors are we expecting here?

Reply:
I believe this was a left over from tests before this bit was had if statement checking for end of vector and was expecting out_of_range, but we are no longer expecting any errors here. Regardless, this part was now changed because of created function (read above) so I removed this try clause.